jvm/ml4ir-inference/src/test/integration/ml4ir/inference/tensorflow/TensorFlowInferenceIT.scala

```diff
@@ -0,0 +1,97 @@
+package ml4ir.inference.tensorflow
+
+import scala.collection.JavaConverters._
+import ml4ir.inference.tensorflow.data.{ModelFeaturesConfig, StringMapExampleBuilder, TestData}
+import org.junit.Test
+import org.junit.Assert._
+import org.tensorflow.example._
+
+import scala.collection.mutable.ListBuffer
+import scala.io.Source
+
+@Test
+class TensorFlowInferenceIT extends TestData {
+  val classLoader = getClass.getClassLoader
+
+  def resourceFor(path: String) = classLoader.getResource(path).getPath
+
+  /**
+   * Class holding a Example and prediction from Python
+   * @param sequence the Example context
+   * @param scores the prediction scores
+   */
+  case class PredictionVector(sequence: Map[String, String], scores: Array[Float])
+
+  /**
+   * Basic parsing of the Python CSV predicition.
+   *
+   * @param line
+   * @return
+   */
+  def extractColumnValues(line: String): Option[PredictionVector] = {
+    val cols = line.split(",").map(_.trim)
+
+    val sequence = Map("query_text" -> cols(1),
+      "domain_id" -> cols(3),
+      "user_context" -> cols(4).substring(1, cols(4).length - 1).trim().replace(" ", ","))
+
+    try {
+      val scores = cols(7).substring(1, cols(7).length - 1).split(" ").map(_.trim).map((s: String) => s.toFloat)
+      Some(PredictionVector(sequence, scores))
+    } catch {
+      case _: NumberFormatException => None
+    }
+  }
+
+  /**
+   * Read a prediction file from the python project.
+   *
+   * @param csvFile path to the CSV
+   * @return List of PredictionVector
+   */
+  def readPredictionCSV(csvFile: String): List[PredictionVector] = {
```

Comment:
so now that the CSV are not full of line-breaks, you can do this method without any mutable state, in one line of scala:

`Source.fromFile(csvFile).getLines.toList.tail.map(extractColumnValues)`

Reply:
although since extractColumnValues returns Option[T] now, maybe that last method would actually be `.flatMap(extractColumnValues)`.  That works, I think.